File: Echowaves/PickWavesForUploadViewController.m
//
//  PickWavesForUploadViewController.m
//  Echowaves
//
//  Created by Dmitry on 8/12/14.
//  Copyright (c) 2014 Echowaves. All rights reserved.
//

#import "PickWavesForUploadViewController.h"

@interface PickWavesForUploadViewController ()

@end

@implementation PickWavesForUploadViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    [EWWave getAllMyWaves:^(NSArray *waves) {
        self.myWaves = waves;
        
        [self.myWavesTableView reloadData];
        [self.myWavesTableView reloadInputViews];
        
    } failure:^(NSError *error) {
        [EWWave showErrorAlertWithMessage:error.description
                               FromSender:nil];
    }];
    
    
}



- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [self.myWaves count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *cellIdentifier = @"wavesPickerCell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellIdentifier];
    
    UILabel* label = (UILabel *)[cell.contentView viewWithTag:500];
    UISwitch* waveOn = (UISwitch *)[cell.contentView viewWithTag:501];
    [label setText:[((NSDictionary*)[self.myWaves objectAtIndex:indexPath.row]) objectForKey:@"name"]];
    NSNumber* isActive = [((NSDictionary*)[self.myWaves objectAtIndex:indexPath.row]) objectForKey:@"active"];
    if (isActive.intValue == 1) {
        waveOn.on = YES;
    } else {
        waveOn.on = NO;
    }
    
    return cell;
}

- (IBAction)waveOnClicked:(id)sender {
    UISwitch *waveOn = sender;

    UITableViewCell *cell = (UITableViewCell *)[waveOn superview].superview;
	if(![cell isKindOfClass:[UITableViewCell class]]) { // ios >= 7.0 http://stackoverflow.com/questions/19162725/access-ios-7-hidden-uitableviewcellscrollview
		cell = (UITableViewCell *)cell.superview;
		assert([cell isKindOfClass:[UITableViewCell class]]);
	}
    UILabel* label = (UILabel *)[cell.contentView viewWithTag:500];
    NSString *waveName = label.text;
    
    [EWWave showLoadingIndicator:self];
    [EWWave makeWaveActive:waveName active:waveOn.isOn
                   success:^(NSString *waveName) {
                       [EWWave hideLoadingIndicator:self];
                   } failure:^(NSString *errorMessage) {
                       [EWWave hideLoadingIndicator:self];
                       [EWWave showAlertWithMessage:errorMessage FromSender:self];
                   }];
}


- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
    return NO;
}

- (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)indexPath {
    return NO;
}

@end

File: Echowaves/PickWavesForUploadViewController.swift
//
//  PickWavesForUploadViewController.swift
//  Echowaves
//
//  Created by D on 12/9/14.
//  Copyright (c) 2014 Echowaves. All rights reserved.
//

import Foundation

class  PickWavesForUploadViewController : UIViewController, UITableViewDelegate, UITableViewDataSource {
    
    
    @IBOutlet var myWavesTableView:UITableView!
    var myWaves = []
    
    @IBOutlet weak var waveNowButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        EWWave.getAllMyWaves(
            { (waves) -> () in
                self.myWaves = waves
                self.myWavesTableView.reloadData()
                self.myWavesTableView.reloadInputViews()
            },
            failure: { (error) -> () in
                EWWave.showErrorAlertWithMessage(error.description, fromSender: self)
        })
        APP_DELEGATE.getPhotosCountSinceLast { (count) -> Void in
            self.waveNowButton.setTitle("Wave \(count) Now", forState: .Normal)
        }
        
    }
    
    @IBAction func waveOnClicked(sender:AnyObject!) {
        let waveOn:UISwitch = sender as UISwitch
        
        var cell:UITableViewCell = waveOn.superview?.superview as UITableViewCell
        let label = cell.contentView.viewWithTag(500) as UILabel
        let waveName = label.text
        
        EWWave.showLoadingIndicator(self)
        EWWave.makeWaveActive(waveName!,
            active: waveOn.on,
            success: { (waveName) -> () in
                EWWave.hideLoadingIndicator(self)
            }, failure: { (errorMessage) -> () in
                EWWave.hideLoadingIndicator(self)
                EWWave.showAlertWithMessage(errorMessage, fromSender: self)
        })
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cellIdentifier = "wavesPickerCell"
        let cell:UITableViewCell? = tableView.dequeueReusableCellWithIdentifier(cellIdentifier) as? UITableViewCell
        
        let label:UILabel = cell?.contentView.viewWithTag(500) as UILabel
        let waveOn:UISwitch = cell?.contentView.viewWithTag(501) as UISwitch
        label.text = ((self.myWaves[indexPath.row] as NSDictionary).objectForKey("name") as String)
        let isActive:Int = ((self.myWaves[indexPath.row] as NSDictionary).objectForKey("active") as Int)
        if (isActive == 1) {
            waveOn.on = true
        } else {
            waveOn.on = false
        }
        return cell!
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.myWaves.count
    }
    
    func tableView(tableView: UITableView, canEditRowAtIndexPath indexPath: NSIndexPath) -> Bool {
        return false
    }
    
    func tableView(tableView: UITableView, canMoveRowAtIndexPath indexPath: NSIndexPath) -> Bool {
        return false
    }
    
}
